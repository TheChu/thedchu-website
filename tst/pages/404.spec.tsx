import { render } from '@testing-library/react';
import { useStaticQuery } from 'gatsby';
import React from 'react';
import uuid from 'short-uuid';
import IndexPage from '../../src/pages/404';

jest.mock('short-uuid');
(uuid.generate as jest.Mock).mockReturnValue('test-uuid');

(useStaticQuery as jest.Mock).mockReturnValue({
  mastheadImage: {
    localMasthead: {
      childImageSharp: {
        gatsbyImageData: {
          layout: 'fullWidth',
          placeholder: {
            fallback: 'test-base64',
          },
          images: {
            fallback: {
              src: 'test-src',
              srcSet: 'test-srcSet',
              sizes: '100vw',
            },
            sources: [],
          },
          width: 1,
          height: 0.6666666666666666,
        },
      },
    },
  },
  site: { siteMetadata: { title: 'thedchu' } },
});
describe('404 IndexPage', () => {
  it('should match 404 IndexPage snapshot', () => {
    const { asFragment } = render(<IndexPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render navigation bar', () => {
    const { getByRole } = render(<IndexPage />);
    expect(getByRole('navigation')).toBeVisible();
  });

  it('should render a link to home page', () => {
    const { getByRole } = render(<IndexPage />);
    expect(getByRole('link', { name: 'Go Home' })).toBeVisible();
  });
});
