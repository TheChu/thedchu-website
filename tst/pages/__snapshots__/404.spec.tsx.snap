// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 IndexPage should match 404 IndexPage snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-light fixed-top py-3 "
    id="mainNav"
  >
    <div
      class="container"
    >
      <div
        class="navbar-item"
      >
        <ul
          class="navbar-nav mr-auto my-2 my-lg-0"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link nav-link-icon mx-2"
              href="https://instagram.com/thedchu"
            >
              <i
                aria-label="Instagram"
                class="fab fa-instagram"
              />
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link nav-link-icon mx-2"
              href="https://linkedin.com/in/thedchu"
            >
              <i
                aria-label="LinkedIn"
                class="fab fa-linkedin-in"
              />
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link nav-link-icon mx-2"
              href="https://github.com/thechu"
            >
              <i
                aria-label="Github"
                class="fab fa-github"
              />
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link nav-link-icon mx-2"
              href="mailto:chu.david.a@gmail.com"
            >
              <i
                aria-label="Email"
                class="fas fa-envelope"
              />
            </a>
          </li>
        </ul>
      </div>
      <a
        class="navbar-brand navbar-item"
        href="/"
      >
        <svg
          aria-label="DChu"
          class="navbar-brand-logo"
          viewBox="0 0 1000 512"
        >
          <use
            href="test-file-stub#thedchu-logo"
          />
        </svg>
      </a>
      <div
        class="navbar-toggler-container"
      >
        <button
          aria-controls="navbarResponsive"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler navbar-toggler-right collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
      </div>
      <div
        class="collapse navbar-collapse "
        id="navbarResponsive"
      >
        <ul
          class="navbar-nav ml-auto my-2 my-lg-0"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#about"
            >
              About
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#photos"
            >
              Photos
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <header
    class="masthead gbi--1668593287-test-uuid"
    style="background-position: center; background-repeat: no-repeat; background-size: cover; position: relative; opacity: 0.99;"
  >
    <style>
      
          .masthead.gbi--1668593287-test-uuid:before,
          .masthead.gbi--1668593287-test-uuid:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            background-color: #040e18;
            transition: opacity 0.5s ease 250ms;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .masthead.gbi--1668593287-test-uuid:before {
            z-index: -100;
            background-image: url(test-base64);
            
            opacity: 1; 
          }
          .masthead.gbi--1668593287-test-uuid:after {
            z-index: -101;
            
            background-image: url(test-base64);
            
          }
        
    </style>
    <div
      class="container h-100"
    >
      <div
        class="row h-100 align-items-center justify-content-center text-center"
      >
        <div
          class="col-lg-10 align-self-end"
        >
          <h1
            class="text-uppercase text-white font-weight-bold"
          >
            Page not found
          </h1>
          <hr
            class="divider my-4"
          />
        </div>
        <div
          class="col-lg-8 align-self-baseline"
        >
          <p
            class="text-white-75 font-weight-light mb-5"
          >
            Not a valid URL
          </p>
          <a
            class="btn btn-primary btn-xl js-scroll-trigger"
            href="/"
          >
            Go Home
          </a>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
